src/pages/blogView/blogView.component.tsx

```diff
@@ -0,0 +1,83 @@
+import { useContext, useEffect, useState } from "react"
+
+import { useLocation, useNavigate } from "react-router-dom"
+import { Bars } from 'react-loader-spinner'
+
+import { ThemeContext } from "../../context/theme/theme.context"
+
+import { getBlogByID } from "../../utils/firebase/firebaseDB.utils"
+
+const BlogView = () => {
+     const { theme } = useContext(ThemeContext);
+     const location = useLocation()
+     const navigate = useNavigate();
+
+     const [title, setTitle] = useState<string>("")
+     const [content, setContent] = useState<string>("")
+     const [email, setEmail] = useState<string>("")
+     const [date, setDate] = useState<string>("")
+     const [loaderVisible, setLoaderVisible] = useState<boolean>(false);
+
+
+     const backButtonClickHandler = () => {
+          navigate(-1);
+     }
+
+     useEffect(() => {
+          const getBlogData = async () => {
```

Comment:
It's better to declare functions outside useEffect and just call them within useEffect.
Also, the content of this function could be broken down to create at least two smaller functions. I suggest dividing the if else code into separate functions.

Reply:
WIth a single state variable, each if/else code block is just 2-3 lines. I think it'd be good enough as it is. 

However, I'll make the changes if you still suggest to divide it into functions.